src/connectiondataholder.cpp

```diff
@@ -224,14 +233,29 @@ void ConnectionDataHolder::updateIpAddress() {
   connect(request, &NetworkRequest::requestCompleted,
           [this](QNetworkReply*, const QByteArray& data) {
             logger.log() << "IP address request completed";
-
             QJsonDocument json = QJsonDocument::fromJson(data);
             if (json.isObject()) {
               QJsonObject obj = json.object();
 
+              QJsonValue country = obj.take("country");
+              if (!m_retryIssued &&
```

Comment:
you need to reset m_retryIssued. Or maybe you can simply get rid of this variable in this way:
```
if (m_checkStatusTimer.isActive() && country.toString().toLower() !=
                      MozillaVPN::instance()->currentServer()->countryCode()) {
```

Reply:
I'm resetting this in ConnectionDataHolder::reset()  - wouldn't m_checkStatusTimer make it a bit racy? In case e.g. the first request takes longer to complete and the user navigates to the page?

Reply:
m_checkStatusTimer is activated when the info page is shown and it's stopped when the user leaves it. I see the following scenarios:
- the IP check is faster than the navigation and good -> all good!
- the IP check is faster than the navigation but with a failure -> in 3 seconds we try again
- - the IP check is slower than the navigation (at the first shot or the following one) -> when completed, we evaluate the next move.

What do you think?